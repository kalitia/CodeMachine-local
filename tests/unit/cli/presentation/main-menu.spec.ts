import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';

// Mock auth status to control login/logout label deterministically.
vi.mock('../../../../src/app/services/auth-status.js', () => {
  return {
    nextAuthMenuAction: vi.fn().mockResolvedValue('login'),
  };
});

describe('renderMainMenu', () => {
  let renderMainMenu: (typeof import('../../../../src/cli/presentation/main-menu.js'))['renderMainMenu'];
  let authModule: any;

  beforeEach(async () => {
    // Load modules fresh for each test so mocks apply
    const mod = await import('../../../../src/cli/presentation/main-menu.js');
    renderMainMenu = mod.renderMainMenu;
    authModule = await import('../../../../src/app/services/auth-status.js');
  });

  afterEach(() => {
    vi.resetModules();
  });

  it('renders the branded main menu with login action', async () => {
    (authModule.nextAuthMenuAction as any).mockResolvedValue('login');
    const output = await renderMainMenu();
    const normalized = output.replace(/\u001b\[[0-9;]*m/g, '');
    expect(normalized).toMatchSnapshot();
  });

  it('renders the branded main menu with logout action', async () => {
    (authModule.nextAuthMenuAction as any).mockResolvedValue('logout');
    const output = await renderMainMenu();
    const normalized = output.replace(/\u001b\[[0-9;]*m/g, '');
    expect(normalized).toMatchSnapshot();
  });
});
