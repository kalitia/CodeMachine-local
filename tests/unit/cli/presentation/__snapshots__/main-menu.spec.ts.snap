// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renderMainMenu > renders the branded main menu with login action 1`] = `
"════════════════════════════════════════════════════════════
 CodeMachine - CLI Coding Agent Specification 
════════════════════════════════════════════════════════════
  ██████╗ ██████╗ ███████╗███╗   ███╗██╗███╗   ██╗██╗
  ██╔══██╗██╔══██╗██╔════╝████╗ ████║██║████╗  ██║██║
  ██████╔╝██████╔╝█████╗  ██╔████╔██║██║██╔██╗ ██║██║
  ██╔═══╝ ██╔══██╗██╔══╝  ██║╚██╔╝██║██║██║╚██╗██║██║
  ██║     ██║  ██║███████╗██║ ╚═╝ ██║██║██║ ╚████║██║
  ╚═╝     ╚═╝  ╚═╝╚══════╝╚═╝     ╚═╝╚═╝╚═╝  ╚═══╝╚═╝
STATUS
────────────────────────────────────────────────────────────
Mode:                   build
COMMANDS
────────────────────────────────────────────────────────────
/start:                 Plan and build from specifications
/templates:             Browse available templates
/login:                 Authenticate with Codex services
/version:               Show CLI version
/help:                  Show command help
/mcp:                   Manage MCP tools and connections
SPECIFICATIONS
────────────────────────────────────────────────────────────
Have you written the full specification in .codemachine/inputs/specifications.md?
Add any necessary context files, then run /start to begin."
`;

exports[`renderMainMenu > renders the branded main menu with logout action 1`] = `
"════════════════════════════════════════════════════════════
 CodeMachine - CLI Coding Agent Specification 
════════════════════════════════════════════════════════════
  ██████╗ ██████╗ ███████╗███╗   ███╗██╗███╗   ██╗██╗
  ██╔══██╗██╔══██╗██╔════╝████╗ ████║██║████╗  ██║██║
  ██████╔╝██████╔╝█████╗  ██╔████╔██║██║██╔██╗ ██║██║
  ██╔═══╝ ██╔══██╗██╔══╝  ██║╚██╔╝██║██║██║╚██╗██║██║
  ██║     ██║  ██║███████╗██║ ╚═╝ ██║██║██║ ╚████║██║
  ╚═╝     ╚═╝  ╚═╝╚══════╝╚═╝     ╚═╝╚═╝╚═╝  ╚═══╝╚═╝
STATUS
────────────────────────────────────────────────────────────
Mode:                   build
COMMANDS
────────────────────────────────────────────────────────────
/start:                 Plan and build from specifications
/templates:             Browse available templates
/logout:                Sign out of Codex services
/version:               Show CLI version
/help:                  Show command help
/mcp:                   Manage MCP tools and connections
SPECIFICATIONS
────────────────────────────────────────────────────────────
Have you written the full specification in .codemachine/inputs/specifications.md?
Add any necessary context files, then run /start to begin."
`;
